### PR TITLE
[CHE-116] Update Default Sorts To Most Recent First

### PR DESCRIPTION
**Pull request by brok3turtl3**
### Description

As part of the [CHE-110](https://codehammers.atlassian.net/browse/CHE-110?atlOrigin=eyJpIjoiOTMzYmQ5ODA0MTJlNDg4ZDlmY2RkYTQ4YWRiOWQyMmUiLCJwIjoiaiJ9) Forums Improvements story this PR adds sort methods to the database calls for threads and posts ensuring they display most recent entries at the top of the feed.

### Jira Task

[CHE-116](https://codehammers.atlassian.net/browse/CHE-116?atlOrigin=eyJpIjoiNGFkMzVjMTFlNTBmNDkyZWEwZTQxYzE5NjBmZTQ1MmIiLCJwIjoiaiJ9)

### Testing Instructions

1. Make sure branches are up to date and spin up dev server off feature branch.
2. Make sure entries are sorted in the expected order on All Forums, a specific forum, and inside a specific thread.

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [ ] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-test` in my local environment to check that this PR passes all unit tests.
- [x] I did a quick check to make sure my code changes follow the recommended style guide.

### Additional Notes, Images, etc.

N/A

[CHE-110]: https://codehammers.atlassian.net/browse/CHE-110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CHE-116]: https://codehammers.atlassian.net/browse/CHE-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/controllers/threadController.ts
Comment (jimpulsee) [inline server/controllers/threadController.ts:1]: Lots of receptive code. Can it be a separate middleware function in the chain?

**Review by jimpulsee — CHANGES_REQUESTED** [review]

**Review by serenahromano — APPROVED** [review]
it looks good to me but you can also make Jimmys changes if you want! but I'm good with keeping it like this

**Review by seantokuzo — APPROVED** [review]
This works for now but I 2nd what Jimmy says. Maybe a new ticket to refactor into some kind of reusable factory fx that handles the sorting and populating

File: server/controllers/forumController.ts [hunk 1]
```diff
@@ -62,10 +62,9 @@ const getForumById = async (
       return res.status(404).json({ message: "Forum not found" });
     }
 
-    const threads = await Thread.find({ forum: forumId }).populate(
-      "user",
-      "firstName lastName"
-    );
+    const threads = await Thread.find({ forum: forumId })
+      .sort({ createdAt: -1 })
+      .populate("user", "firstName lastName");
```
Comment (Mnelson98) [inline server/controllers/forumController.ts]: Looks good.

**Review by Mnelson98 — APPROVED** [review]
Looks good, nice job!

File: server/controllers/threadController.ts
Comment (brok3turtl3) [reply server/controllers/threadController.ts:1]: I'm not sure if that would be worth it for these instances? However, developing a piece of middleware that dynamically handles all sorts and filters sounds interesting.

I'll add a ticket.

Comment (jimpulsee) [reply server/controllers/threadController.ts:1]: Not just sorting but the following logic is repeated multiple times. 

```
      .sort({ createdAt: -1 })
      .populate("user", "firstName lastName")
      .exec();
```

**Review by jimpulsee — APPROVED** [review]

File: server/controllers/threadController.ts
Comment (brok3turtl3) [reply server/controllers/threadController.ts:1]: Yeah I figured that would get encapsulated in the sort helper. I pushed some new code up - This what you had in mind? I was curious about doing it so didn't bother making a new ticket lol

**Review by Mnelson98 — APPROVED** [review]
Changes look good to me.

File: server/controllers/threadController.ts
Comment (jimpulsee) [reply server/controllers/threadController.ts:1]: Something like that, I was thinking more of middleware but a util function like you did is good! Would we need to declare the util function as an async function if we are `await`-ing it?

Comment (brok3turtl3) [reply server/controllers/threadController.ts:1]: I don't think so. Using the exec() method on the end of the query returns the promise directly so then the outer function can handle the `await` -ing part.

**Review by jimpulsee — APPROVED** [review]

**Pull request merged**